activesupport/test/message_verifier_test.rb

```diff
@@ -16,18 +16,133 @@ def load(value)
   end
 
   def setup
-    @verifier = ActiveSupport::MessageVerifier.new("Hey, I'm a secret!")
+    @secret = "Hey, I'm a secret!"
+    @verifier = ActiveSupport::MessageVerifier.new(@secret)
+    @payload = 'data'
+    @options = { expires_at: Time.local(2022), for: 'test' }
+    @claims = ActiveSupport::Claims.new(payload: @payload, **@options)
+  end
+
+  def test_valid_message
+    header, claims, digest = @verifier.generate(@payload, @options).split(".")
+    assert_not @verifier.valid_message?(nil)
+    assert_not @verifier.valid_message?("")
+    assert_not @verifier.valid_message?("\xff") # invalid encoding
+    assert_not @verifier.valid_message?("#{header.reverse}.#{claims}.#{digest}")
+    assert_not @verifier.valid_message?("#{header}.#{claims}.#{digest.reverse}")
+    assert_not @verifier.valid_message?("#{header}.#{claims.reverse}.#{digest}")
+    assert_not @verifier.valid_message?("purejunk")
+    assert_not @verifier.valid_message?("..")
+    assert_not @verifier.valid_message?("pure.junk.data")
+  end
+
+  def test_simple_round_tripping
+    message = @verifier.generate(@payload, @options)
+    assert_equal @payload, @verifier.verified(message, for: 'test')
+    assert_equal @payload, @verifier.verify(message, for: 'test')
+  end
+
+  def raise_invalid_signature_when_value_is_nil
```

Comment:
@kaspth I don't think you got my point. We are testing for nil values of signed messages, not the value to be signed itself. The above discussion is about passing ```nil``` to ```valid_message?``` message and not ```generate``` function, the one I am talking about :smile: 
   As far as backwards compatibility is concerned, I think it means that we should respect all the signed messages from previous(or rather the one in use) ```MessageVerifier``` which we will do. I am just saying that we might want to consider accepting ```nil``` as a value argument for ```generate``` function. It doesn't make sense to sign a ```nil``` value, which our ```MessageVerifier``` does. I hope this makes sense :smile: